scripts/west_commands/runners/sftool.py

```diff
@@ -74,13 +85,18 @@ def do_run(self, command: str, **kwargs):
         if self._erase:
             self.check_call(cmd + ["erase_flash"])
 
+        flash_targets: list[str] = []
+        flash_targets.extend(self._flash_files)
+
         if self.cfg.file:
-            self.check_call(cmd + ["write_flash", self.cfg.file])
+            flash_targets.append(self.cfg.file)
         else:
             if self.cfg.bin_file and self._dt_flash:
                 addr = self.flash_address_from_build_conf(self.build_conf)
-                self.check_call(cmd + ["write_flash", f"{self.cfg.bin_file}@0x{addr:08x}"])
+                flash_targets.append(f"{self.cfg.bin_file}@0x{addr:08x}")
             elif self.cfg.hex_file:
-                self.check_call(cmd + ["write_flash", self.cfg.hex_file])
+                flash_targets.append(self.cfg.hex_file)
             else:
                 raise RuntimeError("No file available for flashing")
+
+        self.check_call(cmd + ["write_flash"] + flash_targets)
```

Comment:
https://docs.python.org/3/library/subprocess.html#subprocess.run

> The recommended approach to invoking subprocesses is to use the run() function for all use cases it can handle.